doc/plugin_server_keymanager_awskms.md

```diff
@@ -0,0 +1,54 @@
+# Server plugin: KeyManager "awskms"
+
+The `awskms` key manager plugin leverages the AWS Key Management Service (KMS) to create, maintain and rotate key pairs (as [Customer master keys](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#master_keys), or CMKs), and sign SVIDs as needed, with the private key never leaving KMS.
+
+## Configuration
+
+The plugin accepts the following configuration options:
+
+| Key               | Type   | Required                              | Description                                          | Default                                              |
+| ----------------- | ------ | ------------------------------------- | ---------------------------------------------------- | ---------------------------------------------------- |
+| access_key_id     | string | see [AWS KMS Access](#aws-kms-access) | The Access Key Id used to authenticate to KMS        | Value of AWS_ACCESS_KEY_ID environment variable      |
+| secret_access_key | string | see [AWS KMS Access](#aws-kms-access) | The Secret Access Key used to authenticate to KMS    | Value of AWS_SECRET_ACCESS_KEY environment variable  |
+| region            | string | yes                                   | The region where the keys will be stored             |                                                      |
+| key_prefix        | string | [1] see below                         | A unique prefix per server                           | SPIRE_SERVER_KEY/                                    |   
+
+[1] `key_prefix` is **optional** when running a single server. When running more than one server on the same region, the prefix **must be set and must be different** on each one. This is a common scenario when running in HA mode. A valid key prefix can be any arbitrary string that uniquely identifies a server instance, like `SERVER_A/` or `prod-server-1/`. It defaults to `SPIRE_SERVER_KEY/` if not specified.
```

Comment:
Woof. This is a sharp edge. I'm not sure what to do about it though. It would be nice if each HA server maintained its own unique identifier, but that isn't currently the case.

@evan2645, curious to hear your thoughts on this.

Reply:
Yea... it's definitely not ideal. One thing that occurred to me is that if the plugin knew which key core was looking for (e.g. via pubkey), then we presumably wouldn't need this configurable prefix at all... or it could be truly optional, on a per-cluster basis rather than a per-server basis.

I reviewed a portion of the KMS documentation and APIs, and I wasn't able to come up with a better solution that doesn't require an interface change :(

Luckily, it looks like core will detect a pubkey mismatch on boot, but we should probably double-check that. This will definitely be error prone.

Reply:
> One thing that occurred to me is that if the plugin knew which key core was looking for (e.g. via pubkey), then we presumably wouldn't need this configurable prefix at all... or it could be truly optional, on a per-cluster basis rather than a per-server basis.

In that case it would be a prefix per cluster. On each server we would fetch all the keys that belong to the cluster, but identify the server keys by the pubkeys that are presented to the API? 

Isn't that conflicting with [this proposed solution?](https://github.com/spiffe/spire/issues/1840) In 1840 we would rely on the keymanager state to identify which certificates belong to the server. While here we would rely on the certificates to infer which keys belong to the server keymanager. I understand those certificates can come from `journal.pem` too but that is also what we are trying to move away from.

> Luckily, it looks like core will detect a pubkey mismatch on boot, but we should probably double-check that. This will definitely be error prone.

[This is where it happens](https://github.com/spiffe/spire/blob/master/pkg/server/ca/manager.go#L627) and I can confirm it works.